ports/espressif/bindings/espnow/Communicate.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * This file is part of the CircuitPython project, https://github.com/adafruit/circuitpython
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2023 MicroDev
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/objproperty.h"
+#include "py/runtime.h"
+
+#include "bindings/espnow/Peer.h"
+#include "common-hal/espnow/__init__.h"
+
+// --- Send and Receive ESP-NOW data ---
+
+//| class Communicate:
```

Comment:
Why have this? This feels weird. Are you trying to abstract for sharing the success and failure statistics? Its probably clearer to just duplicate them in ESP-Now.

Reply:
This allows the following:
```py
e.send()
e.send.success
e.send.failure
```

Reply:
I think that it'll be weird for folks to have attributes on something callable. How about `e.send_stats` and `e.receive_stats` that are tuples of `(success, failure)`? Or you could have 4 properties on e.

Reply:
@MicroDev1 What do you think about simplifying this? It'd be awesome to get this merged so it can be in 8.1

Reply:
Hey, sorry for the late reply, the CI stuff took precedence. 

I like this over adding separate properties in `ESPNow`, it keeps the communication part isolated the same way in which the peers management part is isolated and maintains hierarchy in the module. 

Also, the stats are an important part for making the communication work effectively with minimal packet drops so a no to tuples which as you mentioned earlier aren't self documenting.

Lastly, I don't find it weird to have properties in a callable object, it is python after all.

Reply:
@dhalbert and @jepler Please weigh in on this API design.

Reply:
I feel like it's unusual and so it may be unfamiliar to our users as well..

Reply:
I agree that a more straightforward stats interface would be easier to understand. Maybe it goes with my comment about why to use `job`, instead of just `send` and `read` (`receive` ?). I looked briefly at some ESPnow tutorials and didn't see the same kind of thing, but maybe I am missing something.